src/test/java/FakeInputStream.java

```diff
@@ -0,0 +1,28 @@
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+class FakeInputStream extends InputStream {
+    public FakeInputStream(){}
+
+    public void setInput(String input){
```

Comment:
I know this is a test double, but I would encourage you to avoid setters. Instead I would suggest you leverage the constructor for these types of things. You can also move this logic into a well name factory method.

```java
class FakeInputStream extends InputStream {
    public static FakeInputStream fromString(String data) {
        return new FakeInputStream(something);
    }
}
```

However, is it possible that you don't need any of this? Did you look at using `ByteArrayInputStream`? I think something like this might do the same thing.

```java
new ByteArrayInputStream("my input".getBytes());
```